easybuild/tools/config.py

```
@@ -843,6 +845,33 @@ def log_path(ec=None):
     return log_file_format(return_directory=True, ec=ec, date=date, timestamp=timestamp)
 
 
+def error_log_path(ec=None):
+    """
+    Return the default error log path
+
+    This is a path where file from the build_log_path can be stored permanently
+    :param ec: dict-like value that provides values for %(name)s and %(version)s template values
```

Doesn't match what is used. It expects `name` and `version` keys, not templates

---

I used the `%(name)s` and `%(version)s` template strings as default values. I have seen this use in other cases, `log_file_format` function on the same file (they indeed appear in file paths). It seems like a nice way to inform the user about a missing value without a hard failure.

Should we avoid using template names as default values or explain their use in a better way?

---

Ah I see. Isn't trivial to understand for me what is meant by that.   I'd change the documentation to
>     :param ec: dict-like value with at least the keys 'name' and 'version'

I would not use a default and just fail hard if the keys do not exist because that is what is documented: The function expects a dict with those keys. If you don't provide one it is the fault of the caller. If we do it right in easybuild the keys will always exist, so no need for fallbacks. Or am I missing anything?